Yoast/wordpress-seo
# LINGO-1267 Adjusts the meta description length limit for Japanese

## Context


* In [this PR](https://github.com/Yoast/wordpress-seo/pull/17958), we added fallbacks for the excerpt when that is empty. This PR changes the functionality to adhere to the changes in [this PR](https://github.com/Yoast/wordpress-seo/pull/17907), where we adapted the recommended meta description length (and meta description preview limit) for Japanese. 

## Summary


This PR can be summarized in the following changelog entry:

* Sets the recommended meta description length for Japanese to 80 characters.

## Relevant technical choices:

* We retrieve the locale not from the DOM (as that shows the user's locale, and not the site-wide locale), but from the `l10nObject` (in `packages/js/src/analysis/getContentLocale.js`). 
* The behaviour of live-updating the generated excerpt differs between new and existing posts. For new posts, the excerpt is updated while writing, but for existing posts, the excerpt is only updated after saving. 
* The auto-generated excerpt does not take into account the combined length of the date and separator (visible in the Google Preview for post and pages, though not for taxonomies), while the assessment does take this date into account. This will be addressed in [a separate issue](https://yoast.atlassian.net/browse/LINGO-1243).
* Unit tests were not yet added for changes made inside `seo-integration` package in this PR. This is because no testing framework has been set up for this package. [An issue](https://yoast.atlassian.net/browse/LINGO-1270) has been created to add one.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

**Test the behaviour for Japanese**
- Set a default title and meta description for Posts and Pages:
    - Go to `Yoast SEO > Search Appearance > Content Types`.
    - For both Posts and Pages, set the default meta description to `%%Excerpt%%`.
- Set your site language to Japanese.
- Activate the Classic editor
- Add a new post.
- Add some Japanese text, generate it through [this tool](https://generator.lorem-ipsum.info/_japanese).
- Save the post. 
- Confirm the meta description (in the Google Preview tab) displays the first 80 characters of the post's text. You can use [this tool](https://wordcounter.net/character-count) to check that.
- Add some text **without spaces** to the first 80 characters (e.g. add `testing`). 
- Save the post. 
- Confirm the meta description displays the first 80 characters of the post's text (the meta description should now include your added characters).
- Repeat the above steps for a category.

**Test the default behaviour (regression)**
- Set your site language to English.
- Activate the Classic editor
- Add a new post.
- Add some lorem ipsum text, generate it through [this tool](https://generator.lorem-ipsum.info/).
- Save the post. 
- Confirm the meta description (in the Google Preview tab) displays the first 156 characters of the post's text, stripped at the last space before 156 characters. You can use [this tool](https://wordcounter.net/character-count) to check that.
- Add some text to the first 156 characters (e.g. add `this is a test`). 
- Save the post. 
- Confirm the meta description displays the first 156 characters of the post's text, stripped at the last space before 156 characters (the meta description should now include your added characters).
- Repeat the above steps for a category.

### Test instructions for QA when the code is in the RC



## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

* Affects the meta description replacement variable.

## UI changes


## Other environments

* [ ] This PR also affects Shopify. I have added a changelog entry starting with `[Shopify]` and I have added test instructions for Shopify.

## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-1267
